packages/router/src/grpc/view-protocol-server/transaction-planner.ts

```diff
@@ -28,10 +48,27 @@ export const transactionPlanner: Impl['transactionPlanner'] = async (req, ctx) =
   if (req.memo) planner.memo(req.memo);
   if (req.fee) planner.fee(req.fee);
 
-  for (const { value, address } of req.outputs)
-    if (value && address) planner.output(value, address);
+  for (const { value, address } of req.outputs) {
+    if (!value || !address) throw new Error('no value or address in output');
+    planner.output(value, address);
+  }
+
+  for (const { value, targetAsset, fee, claimAddress } of req.swaps) {
+    if (!value || !targetAsset || !fee || !claimAddress)
+      throw new Error('no value, targetAsset, fee or claimAddress in swap');
+
+    const intoDenomMetadata = await getDenomMetadata(indexedDb, targetAsset, querier);
+    planner.swap(value, intoDenomMetadata, fee, claimAddress);
+  }
+
+  for (const { swapCommitment } of req.swapClaims) {
+    if (!swapCommitment) throw new Error('no swapCommitment in swapClaim');
+    await planner.swapClaim(swapCommitment);
```

Comment:
Interesting... so, we can plan the swap and the swap claim at the same time, even though they'll be in separate transactions/blocks?

Reply:
i don't know about the surrounding context, but in general, planning the swap claim requires knowledge of the `BatchSwapOutputData` for the block the swap was submitted in, which arrives in the `CompactBlock` we use to detect when the swap was committed on-chain. so the swap claim can't be planned at the same time as the swap but it can be planned as soon as the swap is detected

Reply:
this would only attempt to plan a swap and swapClaim at the same time if the caller requests that a swap and swapClaim be planned at the same time by including both in the TransactionPlannerRequest.

the caller probably does not have the information to make such a request, unless they can independently plan and build transactions, and write to idb. but it's possible by pb spec.

Reply:
i had a comment here about why it isn't possible to make such a request but then accidentally deleted it just now, lol. but

1. it's valid pb to make such a request
2. caller doesn't have the information to do so unless they can plan/build independently
3. such a request would likely fail in wasm, which queries idb for that swap commitment, which should not be present yet